### DISCUSSION TITLE
Please add support for Lenovo creator extreme

### DISCUSSION THREAD

**Discussion by urmelchen2705** [MonitorControl/MonitorControl#953] [General]
### Before opening the issue, have you...?

- [X] Searched for existing issues
- [X] Read through [the Monitor-Troubleshooting Wiki](https://github.com/MonitorControl/MonitorControl/wiki/Monitor-Troubleshooting)
- [X] Updated MonitorControl to the latest version (if applicable)

### Describe the issue

DDC not working on Lenovo creator extreme 

### Expected behavior

Control monitor with DDC

### Environment Information (please complete the following information)

```markdown
- macOS version:
- Mac model:
- MonitorControl version:
- Monitor(s):
- Monitor Cable(s)/Connection(s):
- Apple Silicon/M1 (yes or no): yes
```

**Comment by the0neyouseek** [comment]
Hi @urmelchen2705

Please fill out the information about your environment so that we can help debug the issue more effectively.
Could you try using https://github.com/waydabber/m1ddc and see if if works in controlling your monitor ?

**Comment by waydabber** [comment]
Also if you are using HDR please try it in SDR mode and if possible with a standard color mode set in the display. I am interested in this display as I was considering buying one so it would be great to know if it works or not. :)

**Comment by waydabber** [comment]
Also, on Lenovo displays you can long press the Exit button (for 10 secs) to enable/disable DDC/CI so that might help. Please don't use HDMI to connect the display to the Mac as that probably won't work.

**Comment by urmelchen2705** [comment]
Hi 
Ok I will try that later today. The monitor is connected over usb-c

Von meinem iPhone gesendet

**Comment by urmelchen2705** [comment]
I get this:

***@***.*** m1ddc % ./m1ddc display list           
1 - CreatorXtreme (30AEA662-0000-0000-081F-0104B53C2278)

***@***.*** m1ddc % ./m1ddc set luminance 50
I2C communication failure: (iokit/?) unknown subsystem error

**Comment by urmelchen2705** [comment]
DDC was on. No luck yet. 
M1dcc give me:

***@***.*** m1ddc % ./m1ddc display list              
1 - CreatorXtreme (30AEA662-0000-0000-081F-0104B53C2278)

***@***.*** m1ddc % ./m1ddc display 1 set luminance 50
I2C communication failure: (iokit/?) unknown subsystem error

**Comment by waydabber** [comment]
If DDC is enabled in the display and you get this error then something is wrong with the cable or connection maybe (or this display does not support DDC via the USB-C input - which would be strange as it uses displayport alt mode that carries ddc i2c commands all right). It might be that m1ddc is not fully compatible with the new MBP, I didn't try that. You might want to try a simple USB-C to DisplayPort cable if you have one around to see if it works. You can also try other apps like [Lunar](https://lunar.fyi) (@alin23) and see if that works.

Is the display connected directly via a standard USB-C cable provided by the manufacturer?

**Comment by alin23** [comment]
Me and @urmelchen2705 have been talking over email. He has already tried Lunar and has the same issue.

The monitor blocks DDC for some reason.

I took a look at the monitor manual and user guide and these are the settings that need to be changed for brightness control to be allowed:
   
* Disable **Local Dimming**
* Disable **HDR**
* Disable Dynamic **Contrast**
* Set **Color Mode** to Custom
* Set **Scenario Modes** to Panel Native

If those settings are already in that state, then there's nothing we can do to make it work.

**Comment by urmelchen2705** [comment]
All settings are in that state but nothing. I’ve tried two original usb-c cables too

**Comment by alin23** [comment]
I took a look through [the monitor database](https://db.lunar.fyi/) but you seem to be the only one using this monitor with Lunar so far. I have no data on whether the monitor should really support DDC or not. It's possible this is an incompatibility with the M1 DCP (Display Coprocessor) firmware, and hopefully it will be fixed in later updates of macOS.

<img width="1791" alt="image" src="https://user-images.githubusercontent.com/3819725/154133052-d4628c5f-20d8-46d6-a4c5-a07260d82ae8.png">

**Comment by waydabber** [comment]
Oh that's sad as this is a really great display (one that I was considering buying).

But: since this is a FALD/mini-led display, what happens if you use software dimming? If the mini-led control is implemented properly, it should have the same or even superior effect than brightness control via DDC, the hardware backlight should gracefully decrease in intensity while clarity should be retained (somewhat similar to what OLEDs do) so it actually might not make sense to use DDC brightness control anyway. What is your experience with this @urmelchen2705 ? Obviously volume control won't work with DDC being blocked.

**Comment by urmelchen2705** [comment]
Quick Shade from Mac Appstore works great. I’m using this now.

Monitor Control or Lunar doesn’t because it kills the HDR effect

**Reply by alin23** [reply]
> Quick Shade from Mac Appstore works great. I’m using this now. Monitor Control or Lunar doesn’t because it kills the HDR effect

@urmelchen2705 in case others see this, it's worth noting that **QuickShade doesn't use DDC, just a dark overlay over the whole screen**.

Both Lunar and MonitorControl have the same feature as a fallback for monitors that don't respond to DDC controls.

### Forcing Software Overlay dimming in Lunar

<img width="1062" alt="image" src="https://user-images.githubusercontent.com/3819725/154296831-cbe14d27-bb19-43db-a63f-38d275d5f89e.png">

### Forcing Software Overlay dimming in MonitorControl

<img width="842" alt="image" src="https://user-images.githubusercontent.com/3819725/154297974-ca9fd91c-01ca-44c3-ac34-53d39184653b.png">

**Comment by urmelchen2705** [comment]
Yes the Phillips could be good too. But unavailable yet.

I got the Lenovo monitor for 999 Euro

**Reply by waydabber** [reply]
Wow, where did you get this for EUR 999? That is a great price for this display!

**Comment by waydabber** [comment]
Hi @urmelchen2705 - if you enable `Avoid gamma table manipulation` under `Displays` in MonitorControl, the app will use the same software technique as QuickShade for dimming. I believe there might be a similar setting in Lunar as well somewhere.

**Comment by waydabber** [comment]
... you need to disable `Use hardware DDC control` as well for that to work under `Displays`.

**Comment by waydabber** [comment]
fyi - turns out somebody else had an issue with this display as well in the past:

https://github.com/MonitorControl/MonitorControl/issues/514

**Comment by urmelchen2705** [comment]
https://www.heinzsoft-shop.de/

But it is sold out.

Von meinem iPhone gesendet

**Comment by waydabber** [comment]
Well, I have my Lenovo Creator Extreme. I can confirm that DDC-CI does not work at all via USB-C. It works however via DisplayPort.

The screen quality is impressive on the one hand but its software is extremely buggy, slow, the OSD has typos and not-quite-English phrases + it has a horrible coil whine whenever I connect USB-C and try to charge the MacBook. Also it has difficulties changing inputs. So it's a totally mixed bag, I am considering returning it, but will give it a day or two to see if I can figure out its quirks.

**Reply by the0neyouseek** [reply]
@waydabber FYI I've purchased a [DELL U2723QE](https://www.dell.com/en-us/shop/dell-ultrasharp-27-4k-usb-c-hub-monitor-u2723qe/apd/210-bdpf/monitors-monitor-accessories). It works amazingly well with my m1 macbook pro (charges 90w) and DDC works great. It's not a mini-led (for sure at that price) but an IPS "black" with deeper blacks and that does the job for me.

**Reply by waydabber** [reply]
I wanted to try mini-led and this display seemed the best price-to-performance wise.

Btw, some observations about the Creator Extreme users:

- the coil whine (which is probably just my unit) is mostly gone when I change the DP 1.4 connection to DP 1.2 on USB-C. DP 1.4 has practically zero benefits with this display (should theoretically enable USB 3.0 speeds with 4K over USB-C but it doesn't) so it doesn't matter that much I guess.
- if I use an USB-C to DisplayPort cable or a DisplayPort dongle, DDC does work with the Mac. DDC volume control however does not work (I don't use it anyway so it does not bother me personally, but I mention it FYI).

Testing brightness control with DDC, it seems to me that since that this is a mini-LED display, using DDC to control the backlight actually does not make much sense. The results are much better when only gamma control is used (down to absolute black), the mini-LEDs are dimmed properly, the panel brilliantly handles everything and the display smoothly fades down to absolute black.

**Reply by waydabber** [reply]
All right, disregard the info about the coil wine is happening sometimes and other times not. :) I can't find the logic in it anymore. If some setting is changed, the coils start to whine sometimes and stay that way, other times they don't.

Also, doing more tests, DDC backlight dimming combined with gamma dimming is still better as it helps reduce clouding. So not having DDC over USB-C is definitely a loss with this display.

The bottom line is: the display's panel is great, the general build quility is nice as well but the firmware is simply horrible (like they did not even bother using a spell checker), button controls are unintuitive the coil whine problem is extremely annoying. Lenovo's quality control team was on a vacation when they released this display.

**Reply by waydabber** [reply]
Note: DDC over HDMI works well (+ it gives Dolby Vision on PC). I had a spare USB-C+power+DisplayPort+USB hub laying around (luckily it is 4K and HDR capable and passes through DDC). It seems now that if I use this with the Mac and HDMI with the PC and disregard the built-in USB-C port, everything is working smoothly and also there is no coil wine - I think the whine is related to powering the MacBook. After some reading this seems to be a typical issue with many Lenovo USB-C displays. I'll test it like this for a few days and see if it works out. Mini-LED is definitely great in a lot of ways + sometimes I work late into the night so the ultra-low brightness that I can achieve with it is really a life saver, it would be difficult to return to my trusty old LG 4K.

**Reply by waydabber** [reply]
So I exchanged the display to an other one. And behold, it had the exact same issue - horrible coil whine. So I returned that as well. Then the shop went out of stock. So I think that's it, I'll be looking for an other display.

The image quality of this display is really great and the build is very robust and professional looking. But the sub-par electronics (the super noisy power supply) inside and the super-slow (it takes a considerable time for the display to boot up or even wake from sleep) and faulty firmware ruins the entire experience IMHO.

**Reply by waydabber** [reply]
Ok, I went back to the shop and took the original display with the noise issue as they still had it. Just found no alternative in a reasonable price-range and if I don't use the USB-C port, the noise is not actually noticable (except if I concentrate on it really hard or put my ear close to the display). I was just wasn't able to continue to look at my old LG 4K after the mini LED greatness in this display and the MacBook Pro.

**Reply by the0neyouseek** [reply]
I can confirm the mbp indeed has a gorgeous display and I think the only thing better would be oled but we're not quite there yet.

**Reply by waydabber** [reply]
I was thinking about buying the 27EP950 (or 32EP950) but these OLED displays were simply too expensive and in the end unjustifiable + I use the display in a sunlit room so the 250nits limitation for SDR with these OLED panels was a deal breaker - so I went with the Creator Extreme's mini LED display instead. It's quality is comparable to the MBP. Hopefully in a few years bright OLEDs or microLEDs will be available and we can all enjoy perfect black, high brightness combined great response + refresh rate at the same time. :)
